### PR TITLE
Remove redundant WindowsTargetPlatformVersion declarations

### PR DESCRIPTION
**Pull request by JunielKatarn**
Many projects incode `React.Cpp.props`, making it unnecessary to define `WindowsTargetPlatformVersion` or `WindowsTargetPlatformMinVersion`.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9131)

### CODE REVIEW

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj [hunk 1]
```diff
@@ -13,7 +13,6 @@
     <AppContainerApplication>true</AppContainerApplication>
     <ApplicationType>Windows Store</ApplicationType>
     <ApplicationTypeRevision>10.0</ApplicationTypeRevision>
-    <WindowsTargetPlatformVersion Condition=" '$(WindowsTargetPlatformVersion)' == '' ">10.0.19041.0</WindowsTargetPlatformVersion>
     <WindowsTargetPlatformMinVersion>10.0.17134.0</WindowsTargetPlatformMinVersion>
```
Comment (NickGerleman) [inline vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:16]: Is this min version higher than the others? 🤔

**Review by NickGerleman — APPROVED** [review]

File: vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj (hunk 1 above)
Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:16]: Beats me. If we get lucky, it's a mistake and I'll be able to remove that min too after diving into its history.

Comment (asklar) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:16]: React.Cpp.props should get the SDK values from `Microsoft.ReactNative.WindowsSdk.Default.props`

Comment (NickGerleman) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:16]: Honestly the whole bit of property sheets could use a refresh. E.g. React.cpp.props means "property sheets included in projects internal to the ABI layer.

Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:17]: > React.Cpp.props should get the SDK values from `Microsoft.ReactNative.WindowsSdk.Default.props`

Applying this change.

Comment (JunielKatarn) [reply vnext/Microsoft.ReactNative/Microsoft.ReactNative.vcxproj:17]: > Is this min version higher than the others? 🤔

@asklar introduced it to allow WinUI3 compatibility. I left it as it is, but added a comment explaining the override.

I think we could set it **conditionally** iif `$(UseWinUI3)` is set to `true`.

**Pull request merged**